tests/src/Interop/PInvoke/Structures/PInvoke/AsParamNative.cpp

```diff
@@ -0,0 +1,1229 @@
+    {
+        return FALSE;
+    }
+    //str1.i32 = 256;
+    str1.myDelegate1 = NULL;
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByRefOut12(StructSequentialWithDelegateField* str1)
+{
+    //str1->i32 = 256;
+    str1->myDelegate1 = NtestMethod;
+    return TRUE;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByVal13(IncludeOuterIntergerStructSequential str1)
+{
+    if(!IsCorrectIncludeOuterIntergerStructSequential(&str1))
+    {
+        printf("\tMarshalStructAsParam_AsSeqByVal13:IncludeOuterIntergerStructSequential param not as expected\n");
+        PrintIncludeOuterIntergerStructSequential(&str1, "str1");
+        return FALSE;
+    }
+    str1.s.i = 64;
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByRef13(IncludeOuterIntergerStructSequential* str1)
+{
+    if(!IsCorrectIncludeOuterIntergerStructSequential(str1))
+    {
+        printf("\tMarshalStructAsParam_AsSeqByRef13:IncludeOuterIntergerStructSequential param not as expected\n");
+        PrintIncludeOuterIntergerStructSequential(str1, "str1");
+        return FALSE;
+    }
+    ChangeIncludeOuterIntergerStructSequential(str1);
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByRefIn13(IncludeOuterIntergerStructSequential* str1)
+{
+    if(!IsCorrectIncludeOuterIntergerStructSequential(str1))
+    {
+        printf("\tMarshalStructAsParam_AsSeqByRefIn13:IncludeOuterIntergerStructSequential param not as expected\n");
+        PrintIncludeOuterIntergerStructSequential(str1, "str1");
+        return FALSE;
+    }
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByValOut13(IncludeOuterIntergerStructSequential str1)
+{
+    if(!IsCorrectIncludeOuterIntergerStructSequential(&str1))
+    {
+        printf("\tMarshalStructAsParam_AsSeqByValOut13:IncludeOuterIntergerStructSequential param not as expected\n");
+        PrintIncludeOuterIntergerStructSequential(&str1, "str1");
+        return FALSE;
+    }
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByRefOut13(IncludeOuterIntergerStructSequential* str1)
+{
+    ChangeIncludeOuterIntergerStructSequential(str1);
+
+    return TRUE;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByVal14(StructSequentialWithPointerField str1)
+{
+    if(str1.i32 != (LPINT)(16) || str1.i != 32)
+        return FALSE;
+    str1.i = 64;
+    return TRUE;
+}
+
+extern "C" DLL_EXPORT BOOL __stdcall MarshalStructAsParam_AsSeqByRef14(StructSequentialWithPointerField* str1)
+{
+    if(str1->i32 != (LPINT)(16) || str1->i != 32)
+        return FALSE;
+    else
+    {
+        str1->i32 = (LPINT)(str1->i);
```

Comment:
@AaronRobinsonMSFT @luqunl There's warning here, do you have any idea about how to fix?
`warning C4312: 'type cast': conversion from 'INT' to 'LPINT' of greater size`

Reply:
This is similar to the `size_t` issue. The problem here is that `i32` is a 32-bit int, but we are trying to assign a pointer - which on 64-bit build is 64-bit - hence the warning.

Reply:
Try first casting to a `UINT`

Reply:
@AaronRobinsonMSFT Thanks for your suggestion, I tried UINT but failed with similar issue.
So I tried INT64 as first cast like `str1->i32 = (LPINT)(INT64)(str1->i)`, and it works without warnings, do you think INT64 here is reasonable?